### ISSUE TITLE
Compatibility layer for Gosu 0.7/0.8 interface does not work

### ISSUE THREAD

**Issue by redcodefinal** [gosu/gosu#306]
ian@ian-desktop ~/Documents/gamedev/testbed $ irb
2.2.1 :001 > require 'gosu'
 => true 
2.2.1 :002 > Gosu::Image
 => Gosu::Image 
2.2.1 :003 > window = Gosu::Window.new(400, 400, false)
 => #<Gosu::Window:0x00000001252708 @__swigtype__="_p_Gosu__Window"> 
2.2.1 :004 > image = Gosu::Image.new(window, "./block.png")
NoMethodError: undefined method `to_blob' for #<Gosu::Window:0x00000001252708 @__swigtype__="_p_Gosu__Window">
	from (irb):4:in `initialize'
	from (irb):4:in `initialize'
	from (irb):4:in `new'
	from (irb):4
	from /home/ian/.rvm/rubies/ruby-2.2.1/bin/irb:11:in `<main>'
2.2.1 :005 > image = Gosu::Image.new(window, "block.png")
NoMethodError: undefined method `to_blob' for #<Gosu::Window:0x00000001252708 @__swigtype__="_p_Gosu__Window">
	from (irb):5:in `initialize'
	from (irb):5:in `initialize'
	from (irb):5:in `new'
	from (irb):5
	from /home/ian/.rvm/rubies/ruby-2.2.1/bin/irb:11:in `<main>'
2.2.1 :006 > image = Gosu::Image.new(window, "./block.png")
NoMethodError: undefined method `to_blob' for #<Gosu::Window:0x00000001252708 @__swigtype__="_p_Gosu__Window">
	from (irb):6:in `initialize'
	from (irb):6:in `initialize'
	from (irb):6:in `new'
	from (irb):6
	from /home/ian/.rvm/rubies/ruby-2.2.1/bin/irb:11:in `<main>'
2.2.1 :007 > window.to_blob
NoMethodError: undefined method `to_blob' for #<Gosu::Window:0x00000001252708 @__swigtype__="_p_Gosu__Window">
	from (irb):7
	from /home/ian/.rvm/rubies/ruby-2.2.1/bin/irb:11:in `<main>'
2.2.1 :008 >

**Comment by jlnr** [comment]
:/ Both `Gosu::Window#initialize` and `Gosu::Image.new` should support the old interface (which you are using) and the newer, simplified interface (Gosu 0.9+):

```ruby
Gosu::Window.new(800, 600) # no boolean third parameter, pass 'fullscreen: true` if you need fullscreen
Gosu::Image.new("filename.png") # no window parameter
```

See https://libgosu.org/rdoc/ for details.

In theory, the SWIG-generated Ruby wrapper should automatically choose between the new and the old interface. I don't know why it doesn't—probably a bug in SWIG? The best workaround is of course to use the new interface, and stop passing the `Window` references. I'll keep this bug open, maybe it'll happen in one of my own games and I can track it down.

**Comment by jlnr** [comment]
Fixed in Gosu 0.10.4

**Comment by redcodefinal** [comment]
Yeah I did what you said and it fixed it! Glad to see a Gosu::Window is no longer needed to load images, make my code a whole lot cleaner not having to pass Window around like prison wench.
